src/main/groovy/life/qbic/portal/sampletracking/DependencyManager.groovy

```diff
@@ -103,17 +104,26 @@ class DependencyManager {
      * All Exceptions are handled and logged. Skips use cases where instantiation fails.
      */
     private void setupUseCaseInteractors() {
+        def sampleListPresenter
+        def controlElementsPresenter
+        try {
+            sampleListPresenter = new SampleListPresenter(this.viewModel)
+            controlElementsPresenter = new ControlElementsPresenter(this.viewModel)
+        } catch (NullPointerException e){
+            log.error("Could not setup presenters. NullPointer detected.", e)
+        } catch (Exception e) {
+            log.error("Unexpected exception during presenter setup.", e)
+        }
         try {
             SampleTrackingQueryDataSource trackingInfoCenter = SampleTracker.createSampleTrackingInformation(trackingServices.get(0), this.serviceUser)
-            this.queryInfoInteractor = new QuerySampleTrackingInfo(trackingInfoCenter, this.viewModel as SampleTrackingQueryOutput)
+            this.queryInfoInteractor = new QuerySampleTrackingInfo(trackingInfoCenter, sampleListPresenter, controlElementsPresenter)
```

Comment:
two Presenters? Why? I suggest two make the two use cases explicit

Reply:
So completely separate the use cases with interactor and all the rest?

Reply:
The diagram shows why two.